src/System/Random.hs

```diff
@@ -298,10 +298,10 @@ instance Random Char
 instance Random Bool
 instance Random Double where
   randomR r g = runStateGen g (uniformRM r)
-  random g = runStateGen g (uniformRM (0, 1))
+  random g = runStateGen g uniformDouble01M
```

Comment:
Optimisation, should not change behaviour in any way.

Reply:
Looks like somehow there was a change in behavior, one doctest is failing, but I don't see why it would fail

Reply:
This is because `uniformR (0, 1) == 1 - uniformFloat01M`. I have changed the test accordingly: https://github.com/haskell/random/pull/68/files#diff-911d03db4d77c6ca7a8a6a8865daa8f3L279-R279

```diff
- -- 0.5728354935654512
+ -- 0.42716450643454884
```

where 1 - 0.5728354935654512 = 0.42716450643454884.

Reply:
... so this actually does change behaviour. Strictly speaking, this goes against the Reproducibility guarantee: https://hackage.haskell.org/package/random-1.2.0/docs/System-Random.html#g:8

We could
- (A) make the change anyway
- (B) make the change and release the next version as a major version
- (C) change this call so it preserves behaviour

If (C) is not slower than what is currently there, I'm in favour of (C). I'm going to try it out.

Reply:
See https://github.com/haskell/random/pull/68/commits/2cbfe1bde3b38190976356a9bb7f6948dde306dd. The benchmarks don't show a slowdown.

Reply:
I like the C option too. Thank you for adjusting this.

Reply:
> this actually does change behaviour. 

Infinity handling changes behavior too albeit subtly. `uniformRM (0,+∞) g >> uniformM g` will produce different results before and after this PR. Before Word32/64 was generated for uniformRM after generator state is not changed.

Reply:
Good point. It might be good plan for the next version to be `random-1.3.0`. This would allow us to add other minor, but breaking changes, for example deprecation pragmas for the global `theStdGen` and related functions.

Reply:
I'm happy with (C) here, but do not see any downsides in (B).